packages/hot-updater/src/commands/build:native.ts

```diff
@@ -0,0 +1,297 @@
+import fs from "fs";
+import * as p from "@clack/prompts";
+import { type Platform, getCwd, loadConfig } from "@hot-updater/plugin-core";
+
+import { getPlatform } from "@/prompts/getPlatform";
+
+import path from "path";
+import { nativeFingerprint } from "@/utils/fingerprint";
+import { runNativeBuild } from "@/utils/nativeBuild/runNativeBuild";
+import { printBanner } from "@/utils/printBanner";
+import { getNativeAppVersion } from "@/utils/version/getNativeAppVersion";
+import { ExecaError } from "execa";
+
+export interface NativeBuildOptions {
+  outputPath?: string;
+  interactive: boolean;
+  message?: string;
+  platform?: Platform;
+}
+
+export const nativeBuild = async (options: NativeBuildOptions) => {
+  printBanner();
+
+  const cwd = getCwd();
+
+  // const gitCommit = await getLatestGitCommit();
+  // const [gitCommitHash, gitMessage] = [
+  //   gitCommit?.id() ?? null,
+  //   gitCommit?.summary() ?? null,
+  // ];
+
+  const platform =
+    options.platform ??
+    (options.interactive
+      ? await getPlatform("Which platform do you want to deploy?")
+      : null);
+
+  if (p.isCancel(platform)) {
+    return;
+  }
+
+  if (!platform) {
+    p.log.error(
+      "Platform not found. -p <ios | android> or --platform <ios | android>",
+    );
+    return;
+  }
+
+  const config = await loadConfig({ platform, channel: /* todo */ "DUMMY" });
+  if (!config) {
+    console.error("No config found. Please run `hot-updater init` first.");
+    process.exit(1);
+  }
+
+  const target: {
+    appVersion: string | null;
+    fingerprintHash: string | null;
+  } = {
+    appVersion: null,
+    fingerprintHash: null,
+  };
+
+  // calculate fingerprint of the current file state in the native platform directory
+  {
+    const s = p.spinner();
+    s.start(`Fingerprinting (${platform})`);
+    const fingerprint = await nativeFingerprint(cwd, {
+      platform,
+      ...config.fingerprint,
+    });
+
+    target.fingerprintHash = fingerprint.hash;
+    s.stop(`Fingerprint(${platform}): ${fingerprint.hash}`);
+
+    if (!target.fingerprintHash) {
+      p.log.error(`Failed to calculate fingerprint of ${platform}`);
+      return;
+    }
+  }
```

Comment:
How about using `createFingerprintJson` to update fingerprint.json and start the process? But only when the config.updateStrategy is set to "fingerprint".

https://github.com/gronxb/hot-updater/blob/40d28c2a7dfbf787a3987fa43b946ca275f63411/packages/hot-updater/src/utils/fingerprint/index.ts#L86

Reply:
are there any other scripts to run before native build? syncing fingerprint and expo prebuild are sufficient?

Reply:
Yes, that’s right. The fingerprint is common, and the prebuild is probably only for Expo.